contracts/cw721-base/src/state.rs

```diff
@@ -44,3 +44,10 @@ pub fn increment_tokens<S: Storage>(storage: &mut S) -> StdResult<u64> {
     TOKEN_COUNT.save(storage, &val)?;
     Ok(val)
 }
+
+// indexed map needs function, not const (for now at least)
+pub fn tokens<'a, S: Storage + 'a>() -> IndexedMap<'a, 'a, &'a str, TokenInfo, S> {
+    IndexedMap::<&str, TokenInfo, S>::new(b"tokens")
+        .with_index("owner", b"tokens__owner", |d| d.owner.to_vec())
+        .unwrap()
```

Comment:
Yeah, I actually refactored `with_index` stuff here: https://github.com/CosmWasm/cosmwasm-plus/pull/108 and will merge this in. It makes the checks compile-time, rather than run-time ensuring the same index name is not used twice.

Reply:
This is the current demo on how to construct this - a bit harder but no unwrap and the usage is much cleaner: https://github.com/CosmWasm/cosmwasm-plus/pull/125/files#diff-e914973c827114d0c37ed9b0a9d7b0bb3105fba992674a1cb2b32369fa6d50bfR152-R170